airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -351,9 +351,8 @@ public WebBackendConnectionRead webBackendGetConnection(final WebBackendConnecti
        * but was present at time of configuration will appear in the diff as an added stream which is
        * confusing. We need to figure out why source_catalog_id is not always populated in the db.
        */
-      diff = connectionsHandler.getDiff(catalogUsedToMakeConfiguredCatalog.orElse(configuredCatalog), refreshedCatalog.get().getCatalog(),
-          CatalogConverter.toProtocol(configuredCatalog));
-
+      diff = refreshedCatalog.get().getCatalogDiff();
+      connection.setBreakingChange(refreshedCatalog.get().getBreakingChange());
```

Comment:
Calling `.get()` on an `Optional.empty()` will throw a `NoSuchElementException`.  What is the expected behavior here if `refreshedCatalog = Optional.empty();` (from line 333) was the path taken?

Reply:
on line 339 we have `if(refreshedCatalog.isPresent())` so all of this is nested under that - I think that should make this ok?